### ISSUE TITLE
false positive on pointer passed without &

### ISSUE THREAD

**Issue by jonsyu1** [palantir/outparamcheck#1]
outparam complains about the following code (taken from the usage docs of configloading.SetToDefault):
```
package main

import (
	"github.palantir.build/deployability/skylab/config"
)

func main() {
	conf := new(config.DeploydConfig)
	_ = config.SetToDefaults(conf)
}
```
Expected: no errors, since `conf` is a `*config.DeploydConfig`.
Actual:
```
Running outparamcheck...
github.palantir.build/deployability/skylab/play.go:9:27: _ = config.SetToDefaults(conf)  // 1st argument of 'SetToDefaults' requires '&'
```

**Comment by nmiyake** [comment]
@dtolnay

**Comment by dtolnay** [comment]
Outparamcheck enforces `&` even if the variable is a pointer. Consider:

```go
var conf *config.DeploydConfig
_ = config.SetToDefaults(conf) // this is a bug even though `conf` is a pointer
```

Libraries should be written in a way that an extra layer of pointer doesn't matter. This is true of the json and yaml libraries and hopefully true of SetToDefault - if not, let's fix it.

If you really truly want to pass a value directly from a variable, there is an escape hatch to suppress the check:

```go
f(*&x)
```

But in this case the usage doc should be updated to read:

```go
var conf config.DeploydConfig
_ = config.SetToDefaults(&conf)
```

**Comment by jonsyu1** [comment]
I agree with your example and your stance on libraries handling an extra level of indirection. The escape hatch is reasonable.

The README seems to suggest that outparamcheck only enforces that the argument is a pointer and should be updated to reflect that it enforces `&` on the argument:
> It verifies that functions that take output parameters defined as interface{} types are passed pointers to an object rather than a concrete object
